### ISSUE TITLE
data leakage in "wiki-rolling_nips" dataset

### ISSUE THREAD

**Issue by y-tashi** [awslabs/gluon-ts#1418]
## Description
Dear GluonTs Team,

I used  "wiki-rolling_nips" dataset and I found that train and test dataset are ovelapping. Namely, there seem to be data leakages.

In "gluon-ts/src/gluonts/dataset/repository/_gp_copula_2019.py", rolling_evaluations of "wiki-rolling_nips" is 5, but the 1st evaluation of test data is in train dataset. 

Here is an example. The last 20 records in train and test are same, so test evaluation uses data in training dataset.

last 20 records of item_id 0 in train dataset:
2813.0, 2587.0, 2488.0, 1663.0, 2119.0, 2494.0, 2679.0, 2766.0, 2704.0, 2478.0, 1755.0, 2086.0, 2734.0, 2789.0, 2867.0, 2923.0, 2630.0, 1732.0, 1908.0, 2534.0

last 20 records of item_id 0 in test dataset:
2813.0, 2587.0, 2488.0, 1663.0, 2119.0, 2494.0, 2679.0, 2766.0, 2704.0, 2478.0, 1755.0, 2086.0, 2734.0, 2789.0, 2867.0, 2923.0, 2630.0, 1732.0, 1908.0, 2534.0

## To Reproduce
check files in the following link:
https://github.com/mbohlkeschneider/gluon-ts/raw/650ad5ffe92d20e89d491966b6d8b4459e219be8/datasets/wiki-rolling_nips.tar.gz

**Comment by lostella** [comment]
Here's a snippet to reproduce it, we could add something like this to the tests:

```python
from gluonts.dataset.repository.datasets import get_dataset

dataset = get_dataset("wiki-rolling_nips")

prediction_length = dataset.metadata.prediction_length

train_end = {}
for entry in dataset.train:
    train_end[entry["item_id"]] = entry["start"] + len(entry["target"]) * entry["start"].freq

test_end = {}
for entry in dataset.test:
    test_end[entry["item_id"]] = entry["start"] + len(entry["target"]) * entry["start"].freq

for k in test_end:
    if k not in train_end:
        continue
    end_diff = test_end[k] - train_end[k]
    exp_diff = prediction_length * train_end[k].freq
    assert end_diff >= exp_diff, f"item {k}: diff = {end_diff} < {exp_diff}"
```

Fails with 

```
Traceback (most recent call last):
  File "issues/run_1418.py", line 20, in <module>
    assert end_diff >= exp_diff, f"item {k}: diff = {end_diff} < {exp_diff}"
AssertionError: item 0: diff = 0 days 00:00:00 < <30 * Days>
```

**Comment by lostella** [comment]
A quick check reveals that `wiki-rolling_nips` and `m4_yearly` have issues. Gist here: https://gist.github.com/lostella/ee40ffde4431c5f6f38404b5fb557c47

Output:

```
✓    dataset 'constant' looks good
✓    dataset 'exchange_rate' looks good
✓    dataset 'solar-energy' looks good
✓    dataset 'electricity' looks good
✓    dataset 'traffic' looks good
✓    dataset 'exchange_rate_nips' looks good
✓    dataset 'electricity_nips' looks good
✓    dataset 'traffic_nips' looks good
✓    dataset 'solar_nips' looks good
ERR  dataset 'wiki-rolling_nips' has issues: test entry for item 0 ends at 2014-03-03 00:00:00 < 2014-04-02 00:00:00
✓    dataset 'taxi_30min' looks good
WARN dataset 'm3_monthly' could not be obtained
WARN dataset 'm3_quarterly' could not be obtained
WARN dataset 'm3_yearly' could not be obtained
WARN dataset 'm3_other' could not be obtained
✓    dataset 'm4_hourly' looks good
✓    dataset 'm4_daily' looks good
✓    dataset 'm4_weekly' looks good
✓    dataset 'm4_monthly' looks good
✓    dataset 'm4_quarterly' looks good
ERR  dataset 'm4_yearly' has issues: test entry for item 3819 ends at 2050-12-31 00:00:00 < 2056-12-31 00:00:00
WARN dataset 'm5' could not be obtained
```

**Comment by lostella** [comment]
Fixed in #1445, thanks for catching this @y-tashi! You should be able to force re-downloading the fixed datasets with the `regenerate=True` option:

```
get_dataset(name, regenerate=True)
```
